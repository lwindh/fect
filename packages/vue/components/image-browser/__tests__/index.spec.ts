import { mount } from '@vue/test-utils'
import ImageBrowser from '..'
import Image from '../../image'

const Wrapper = {
  components: {
    [ImageBrowser.name]: ImageBrowser,
    [Image.name]: Image
  },
  data() {
    return {
      invert: false,
      url: 'https://www.miaya.art',
      showFullLink: false,
      title: 'Fect'
    }
  },
  template: `
    <div classs="container">
      <fe-imageBrowser 
       :invert="invert"
       :url="url" 
       :showFullLink="showFullLink" 
       :title="title">
        <fe-image width="540px" height="246px" />
      </fe-imageBrowser>
    </div>
  `
}

describe('ImageBrowser', () => {
  it('should render as a element', () => {
    const wrapper = mount(Wrapper)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be support invert and showFullLink', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ invert: true, showFullLink: true })
    expect(wrapper.find('.fect-image__browser')).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should only render title when without url', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ url: '' })
    expect(wrapper.find('.address-input').exists()).toBe(false)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should render empty when title and url is empty', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ url: '', title: '' })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('if the URL address is incorrect, the original value is displayed', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ url: '123' })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
