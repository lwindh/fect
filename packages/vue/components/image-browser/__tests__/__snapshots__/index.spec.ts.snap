// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBrowser if the URL address is incorrect, the original value is displayed 1`] = `
<div classs="container">
  <div class="fect-image__browser">
    <header class="fect-image__traffic">
      <div class="traffic__content"><span class="close" role="browser-close"></span><span class="mini" role="browser-mini"></span><span class="full" role="browser-full"></span></div>
      <div class="fect-image__input"><span class="https"><svg width="12" height="12" viewBox="0 0 24 24" fill="currentColor"><path d="M5 10.2H19V21H5V10.2Z"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M12 12C13.933 12 15.5 10.3882 15.5 8.4C15.5 6.41177 13.933 4.8 12 4.8C10.067 4.8 8.5 6.41177 8.5 8.4C8.5 10.3882 10.067 12 12 12ZM12 13.8C14.8995 13.8 17.25 11.3823 17.25 8.4C17.25 5.41766 14.8995 3 12 3C9.10051 3 6.75 5.41766 6.75 8.4C6.75 11.3823 9.10051 13.8 12 13.8Z"></path></svg></span><a class="fect-link " href="123">123</a></div>
    </header>
    <div class="fect-image" style="width: 540px; height: 246px; overflow: hidden;"><img src="" width="0" height="0"></div>
  </div>
</div>
`;

exports[`ImageBrowser should be support invert and showFullLink 1`] = `
<div classs="container">
  <div class="fect-image__browser">
    <header class="fect-image__traffic">
      <div class="traffic__content"><span class="close" role="browser-close"></span><span class="mini" role="browser-mini"></span><span class="full" role="browser-full"></span></div>
      <div class="fect-image__input"><span class="https"><svg width="12" height="12" viewBox="0 0 24 24" fill="currentColor"><path d="M5 10.2H19V21H5V10.2Z"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M12 12C13.933 12 15.5 10.3882 15.5 8.4C15.5 6.41177 13.933 4.8 12 4.8C10.067 4.8 8.5 6.41177 8.5 8.4C8.5 10.3882 10.067 12 12 12ZM12 13.8C14.8995 13.8 17.25 11.3823 17.25 8.4C17.25 5.41766 14.8995 3 12 3C9.10051 3 6.75 5.41766 6.75 8.4C6.75 11.3823 9.10051 13.8 12 13.8Z"></path></svg></span><a class="fect-link " href="https://www.miaya.art">https://www.miaya.art</a></div>
    </header>
    <div class="fect-image" style="width: 540px; height: 246px; overflow: hidden;"><img src="" width="0" height="0"></div>
  </div>
</div>
`;

exports[`ImageBrowser should only render title when without url 1`] = `
<div classs="container">
  <div class="fect-image__browser">
    <header class="fect-image__traffic">
      <div class="traffic__content"><span class="close" role="browser-close"></span><span class="mini" role="browser-mini"></span><span class="full" role="browser-full"></span></div>
      <div class="fect-image__title"><span>Fect</span></div>
    </header>
    <div class="fect-image" style="width: 540px; height: 246px; overflow: hidden;"><img src="" width="0" height="0"></div>
  </div>
</div>
`;

exports[`ImageBrowser should render as a element 1`] = `
<div classs="container">
  <div class="fect-image__browser">
    <header class="fect-image__traffic">
      <div class="traffic__content"><span class="close" role="browser-close"></span><span class="mini" role="browser-mini"></span><span class="full" role="browser-full"></span></div>
      <div class="fect-image__input"><span class="https"><svg width="12" height="12" viewBox="0 0 24 24" fill="currentColor"><path d="M5 10.2H19V21H5V10.2Z"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M12 12C13.933 12 15.5 10.3882 15.5 8.4C15.5 6.41177 13.933 4.8 12 4.8C10.067 4.8 8.5 6.41177 8.5 8.4C8.5 10.3882 10.067 12 12 12ZM12 13.8C14.8995 13.8 17.25 11.3823 17.25 8.4C17.25 5.41766 14.8995 3 12 3C9.10051 3 6.75 5.41766 6.75 8.4C6.75 11.3823 9.10051 13.8 12 13.8Z"></path></svg></span><a class="fect-link " href="https://www.miaya.art">www.miaya.art</a></div>
    </header>
    <div class="fect-image" style="width: 540px; height: 246px; overflow: hidden;"><img src="" width="0" height="0"></div>
  </div>
</div>
`;

exports[`ImageBrowser should render empty when title and url is empty 1`] = `
<div classs="container">
  <div class="fect-image__browser">
    <header class="fect-image__traffic">
      <div class="traffic__content"><span class="close" role="browser-close"></span><span class="mini" role="browser-mini"></span><span class="full" role="browser-full"></span></div>
      <!---->
    </header>
    <div class="fect-image" style="width: 540px; height: 246px; overflow: hidden;"><img src="" width="0" height="0"></div>
  </div>
</div>
`;
